packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -710,12 +723,20 @@ export class RestApi extends RestApiBase {
    * Import an existing RestApi that can be configured with additional Methods and Resources.
    */
   public static fromRestApiAttributes(scope: Construct, id: string, attrs: RestApiAttributes): IRestApi {
+    const restApiName = attrs.restApiName ?? '';
+
     class Import extends RestApiBase {
       public readonly restApiId = attrs.restApiId;
       public readonly restApiRootResourceId = attrs.rootResourceId;
       public readonly root: IResource = new RootResource(this, {}, this.restApiRootResourceId);
     }
 
+    class ImportWithApiName extends Import {
+      public readonly restApiName = restApiName;
+    }
```

Comment:
Why not add `restApiName` to `Import` itself, so we don't need this extra class?

Reply:
I was struggling with the fact that the attribute was optional, but I guess we can fall back to `id`

Reply:
Removed the unnecessary class